### ISSUE TITLE
Create and Update functions should not mutate the options parameter

### ISSUE THREAD

**Issue by jitran** [realestate-com-au/stackup#64]
The functions create() and update() are currently mutating the options parameter. Similar to the create_or_update() function, they should duplicate the options parameter into a local variable called options, i.e. `options = options.dup` (https://github.com/realestate-com-au/stackup/blob/master/lib/stackup/stack.rb#L115)

For example, we're invoking the create_or_update() function like so to create a stack:
```
    stack = Stackup.stack(stack_name)
    stack.create_or_update(
      ..
      template: template,
      disable_rollback: true
    )
```

We have a working CloudFormation template and a set of parameters that are expected to fail the stack creation.

Here's a sequence of events that show how a stack creation with disable_rollback=true behaves inconsistently:

1. [SUCCESS] Invoke the code to create a brand new stack. The stack creation fails but as instructed, the resources are not rolled back, and the stack ends with a status of 'CREATE_FAILED'.
1. [FAILED] Invoke the code with the same template and params to create the same stack using the same name. Stackup successfully deletes the existing stack and then proceeds to create a new stack.
   1. This time around the stack creation fails as expected, but all the resources were rolled back, and the stack ends with a status of 'ROLLBACK_COMPLETE'.
   1. Looking through the code, it appears:
      1. L132 `delete if ALMOST_DEAD_STATUSES.include?(status)` was responsible for deleting the existing failed stack from step 1
      1. L133 `update(options)` was executed;
         1. it mutated the options param by removing disable_rollback on L293 `options.delete(:disable_rollback)`
         1. as the stack had just been deleted, a NoSuchStack exception was thrown and caught by L134 of the `create_or_update()` function
         1. `create(options)` is finally executed to create the stack, but this time around, `disable_rollback` is no longer set in the options hash, as it was deleted in the update function, hence the stack creation failure followed by a rollback

Can the create() and update() functions be updated to duplicate the options parameter into a local variable instead?

Thanks

**Comment by ps-jay** [comment]
I assume this has been resolved by https://github.com/realestate-com-au/stackup/pull/65 ?

**Comment by jitran** [comment]
Closing this issue as it's been resolved in #65.
